### PR TITLE
HSEARCH-1815 SerializationProvider documentation

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/HSEARCH-1815

### CODE REVIEW

File: documentation/src/main/asciidoc/configuration.asciidoc [hunk 1]
```diff
@@ -894,7 +894,51 @@ hibernate.search.[default|<indexname>].reader.strategy = my.corp.myapp.CustomRea
 
 where my.corp.myapp.CustomReaderProvider is the custom strategy implementation.
 
+=== Serialization
 
+When using clustering features, Hibernate Search needs to find an implementation of the
+`SerializationProvider` service on the classpath.
+
+An implementation of the service for link:https://avro.apache.org[Apache Avro] can be found
```
Comment (Sanne) [inline documentation/src/main/asciidoc/configuration.asciidoc]: 'for' should probably be "based on" or "using" ?

File: documentation/src/main/asciidoc/configuration.asciidoc [hunk 2]
```diff
@@ -894,7 +894,51 @@ hibernate.search.[default|<indexname>].reader.strategy = my.corp.myapp.CustomRea
...
+    public Deserializer getDeserializer() {
+        Deserializer deserializer = ...
+        return deserializer;
+    }
+}
+----
+====
+
+Once you have an implementation of the service you need to register it.
+If you are not familiar with this step, you will find more details about it in the
```
Comment (Sanne) [inline documentation/src/main/asciidoc/configuration.asciidoc]: Really? Don't we do Service discovery?

File: engine/src/test/java/org/hibernate/search/test/backend/serialization/SerializationProviderMissingTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.rules.ExpectedException;
+
+/**
+ * When Hibernate Search is used in a clustering context, it requires a SerializationProvider.
+ * This test checks that the error message is giving some details to the user if a SerializationProvider is not registered.
+ *
+ * @author Davide D'Alto
+ */
+@TestForIssue(jiraKey = "HSEARCH-1815")
+public class SerializationProviderMissingTest {
```
Comment (Sanne) [inline engine/src/test/java/org/hibernate/search/test/backend/serialization/SerializationProviderMissingTest.java]: Nice! I only expected two lines in the documentation :)

File: documentation/src/main/asciidoc/configuration.asciidoc (hunk 2 above)
Comment (DavideD) [inline documentation/src/main/asciidoc/configuration.asciidoc]: Yeah, but you still need to add the file in the META-INF. I'll rephrase it

Comment (Sanne) [inline documentation/src/main/asciidoc/configuration.asciidoc]: Ok, but that has nothing to do with `org.hibernate.search.engine.service.spi.Service`

Comment (DavideD) [inline documentation/src/main/asciidoc/configuration.asciidoc]: Isn't SerializationProvider a service?
Do you mean that if I add a custom implementation I don't have to add the file under META-INF?

Comment (Sanne) [inline documentation/src/main/asciidoc/configuration.asciidoc]: I don't know which kind it is anymore, you just confused me :)

But the section <<section-services>> of documentation is referring to a specific type of services than the Standard Java ServiceLoader so we should not confuse the implementor.

Comment (DavideD) [inline documentation/src/main/asciidoc/configuration.asciidoc]: Mmmh...I'm not sure where did you see a reference to the standard ServiceLoader.
Anyway, I will change this part.

File: documentation/src/main/asciidoc/configuration.asciidoc (hunk 1 above)
Comment (DavideD) [inline documentation/src/main/asciidoc/configuration.asciidoc]: ok

**Comment by DavideD** [comment]
Remarks applied and rebased

**Comment by Sanne** [comment]
merged, thanks!

**Pull request merged**